modules/tinymce/src/plugins/table/test/ts/browser/NestedFakeSelectionTest.ts

```diff
@@ -0,0 +1,436 @@
+    },
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin, Theme ]);
+
+  // The critical part is the target element as this is what Darwin (MouseSelection.ts) uses to determine the fake selection
+  const selectWithMouse = (start: SugarElement<Element>, end: SugarElement<Element>) => {
+    Mouse.mouseDown(start, { button: 0 });
+    Mouse.mouseOver(end, { button: 0 });
+    Mouse.mouseUp(end, { button: 0 });
+  };
+
+  // Set up to mock what the listeners are looking for in InputHandlers.ts - keyup()
+  const selectWithKeyboard = (editor: Editor, cursorRange: Cursors.CursorPath, direction: number) => {
+    const { startPath, soffset, finishPath, foffset } = cursorRange;
+    TinySelections.setSelection(editor, startPath, soffset, finishPath, foffset);
+    TinyContentActions.keydown(editor, direction, { shiftKey: true });
+    TinyContentActions.keyup(editor, direction, { shiftKey: true });
+  };
+
+  const getSelectedCells = (editor: Editor) =>
+    SelectorFilter.descendants(TinyDom.body(editor), 'td[data-mce-selected],th[data-mce-selected]');
+
+  const assertSelectedCells = (editor: Editor, expectedSelectedCells: string[]) => {
+    const selectedCells = Arr.map(getSelectedCells(editor), (cell) => Attribute.get(cell, 'data-cell'));
+    assert.deepEqual(selectedCells, expectedSelectedCells);
+  };
+
+  const assertMouseTableSelection = (editor: Editor, content: string, selection: Cursors.CursorPath, expectedSelectedCells: string[]) => {
+    editor.setContent(content);
+    const cursorRange = Cursors.calculate(TinyDom.body(editor), selection);
+    selectWithMouse(cursorRange.start, cursorRange.finish);
+    assertSelectedCells(editor, expectedSelectedCells);
+  };
+
+  const assertKeyboardTableSelection = (editor: Editor, content: string, selection: Cursors.CursorPath, direction: number, expectedSelectedCells: string[]) => {
+    editor.setContent(content);
+    selectWithKeyboard(editor, selection, direction);
+    assertSelectedCells(editor, expectedSelectedCells);
+  };
+
+  const assertTableSelection = (editor: Editor, scenario: Scenario) => {
+    assertMouseTableSelection(editor, scenario.content, scenario.selection, scenario.expectedSelectedCells);
+    assertKeyboardTableSelection(editor, scenario.content, scenario.selection, scenario.keyDirection, scenario.expectedSelectedCells);
+  };
+
+  it('TINY-6298: Partially selecting cell content in the nested table', () => {
+    const editor = hook.editor();
+    const content = (
+      '<table>' +
+      '<tbody>' +
+      '<tr>' +
+      '<td data-cell="1a">' +
+      '<p>abc</p>' +
+      '<table>' +
+      '<tbody>' +
+      '<tr>' +
+      '<td data-cell="2a">nested1</td>' +
+      '<td data-cell="2b">nested2</td>' +
+      '</tr>' +
+      '</tbody>' +
+      '</table>' +
+      '<p>def</p>' +
+      '</td>' +
+      '</tr>' +
+      '</tbody>' +
+      '</table>'
+    );
```

Comment:
If we're doing this chopped down approach, can we also add some indentation to make this more readable please?

Here's an example for this table, but I'll leave the rest up to you:
```suggestion
    const content = (
      '<table>' +
        '<tbody>' +
          '<tr>' +
            '<td data-cell="1a">' +
              '<p>abc</p>' +
              '<table>' +
                '<tbody>' +
                  '<tr>' +
                    '<td data-cell="2a">nested1</td>' +
                    '<td data-cell="2b">nested2</td>' +
                  '</tr>' +
                '</tbody>' +
              '</table>' +
              '<p>def</p>' +
            '</td>' +
          '</tr>' +
        '</tbody>' +
      '</table>'
    );
```

Reply:
It might also be worthwhile pulling the table out into a separate variable or function to reduce the duplication/size of this, as most of these are identical or have minor changes.

- [ ] Consider removing the duplication with re-declaring the table html

Reply:
Have moved the main table HTML into a variable and cleaned it up. Didn't turn it into a function as I thought it might make the HTML too difficult to read.

Reply:
Looks good to me thanks!